hubspot/tasks.py

```diff
@@ -68,14 +77,57 @@ def check_hubspot_api_errors():
         defaults={"checked_on": now_in_utc()}
     )
     last_timestamp = hubspot_timestamp(last_check.checked_on)
+
+    order_sync_cache = {}
+
     for error in get_sync_errors():
-        if error.get("errorTimestamp") > last_timestamp:
-            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
-                obj_type=error.get("objectType", "N/A"),
-                obj_id=error.get("integratorObjectId", "N/A"),
-                details=error.get("details", ""),
+        error_timestamp = error.get("errorTimestamp")
+        if error_timestamp > last_timestamp:
+            obj_type = (error.get("objectType", "N/A"),)
+            obj_id = parse_hubspot_id(error.get("integratorObjectId", "")) or "N/A"
+            error_type = error.get("type", "N/A")
+            details = error.get("details", "")
+            # Handle line syncs that may have failed because of pending order syncs
+            if "LINE_ITEM" in obj_type and error_type == "INVALID_ASSOCIATION_PROPERTY":
+                order_id = parse_hubspot_id(ASSOCIATED_DEAL_RE.search(details).group(1))
```

Comment:
Can you create a separate function for the below code, something like `def retry_line_item_sync(line_id, order_id)` that maybe returns a boolean (`True` if another line sync attempt was made, `False` if otherwise). 

```python
if "LINE_ITEM" in obj_type and error_type == "INVALID_ASSOCIATION_PROPERTY":
    order_id = parse_hubspot_id(ASSOCIATED_DEAL_RE.search(details).group(1))
    if retry_line_item_sync(line_id, order_id):
        continue  # Continue here to avoid logging an error
log.error(.....
```